action.yml

```
@@ -5,15 +5,42 @@ branding:
   icon: 'bar-chart'
   color: 'purple'
 
+inputs:
+  command:
+    description: 'The command to run'
+    required: true
+    type: choice
+    options:
+      - 'analyze'
+      - 'track-package'
+      - 'monitor-repo'
```

Can we add a default for the command? That way this update won't be a breaking change for users.
```suggestion
      - 'monitor-repo'
    default: 'analyze'
```

---

I spoke to the team about adding a default this morning. We don't want to have a default here as this action can be run in both DS package repos as well as app repos and having the default of `analyze` could be confusing.

We are happy with having a breaking change here as we are doing a major version bump and it won't break for users using `v1` in their workflows

---

Ah ok, would you be able to share in more detail what all the options do? I read the docs on npm but I don't find them to be very clear and I feel like the different options have different use cases. Thanks!

---

Ah yeah, the docs aren't super clear at the moment sorry

* `track-package` is the same as setting up a package from npm or git in zeroheight, the first step in the current adoption flow.
* `monitor-repo` is the same as selecting repos to monitor versions of the package for in zeroheight
* `analyze` is for tracking component usage, works the same way as it does now in the CLI tool

There are some docs around the new options here: https://zeroheight.com/help/article/package-version-monitoring/ and we are planning on updating the docs + npm readme with more info once we've enabled the new set up flow

We are also currently making a new video like the current one we have to go into more detail about how the new flow works

---

Ah this is much better, thanks! If `track-package` is for setting up a package in the flow I probably wouldn't even allow people to use that option in this action as it's a one time use and would need to be removed after setting up. And I guess it's the same case for `monitor-repo` as it's only ran once to select the repos. Makes me wonder if we should even have these options in this action 🤔. The intended use is to install it in a GitHub project and leave it to automatically provide component data on each deploy of an application

---

Sorry it's not just for setting up.

`track-package` also updates the version in zeroheight to the latest in your design systems `package.json` and `monitor-repo` does the same in the repos that are consuming the package from the `package-lock.json`/`yarn.lock` file similar to how running the update manually in zeroheight also updates the versions

---

After checking out the video Rosie made these features are making a lot more sense! I still think we should have some kind of default for the action though. It makes sense that the CLI tool doesn't have a default, because it could be being used for anything, but for a public GitHub Action it's a lot more likely it's submitting component data than submitting package version data because of our GitHub integrations. I get the concern about confusing options, but I'm worried someone will miss the configuration part and be left with a silently failing pipeline tool. Hope that makes sense

---

> but for a public GitHub Action it's a lot more likely it's submitting component data than submitting package version data because of our GitHub integrations

We added these commands because we had a lot of feedback that teams cannot use the git integrations for security reasons. We're not expecting any one of these commands to get more use than the others at the moment as we're intending them to be part of regular release workflows in different types of projects

> I get the concern about confusing options, but I'm worried someone will miss the configuration part and be left with a silently failing pipeline tool

Good spot! I didn't realise that GitHub actions don't validate `required` inputs if they are completely missing. I have just added validation to the `command` input to ensure it is never left blank by accident